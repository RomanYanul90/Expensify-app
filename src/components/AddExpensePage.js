import React from 'react';
import { connect } from 'react-redux';
import Expense_form from './Expense_form';
import { addExpense } from '../actions/expenses';

export class AddExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.addExpense(expense);
        this.props.history.push('/');
    };
    render() {
        return (
            <div>
                <h1>Add Expense</h1>
                <Expense_form
                    onSubmit={this.onSubmit}
                />
            </div>
        );
    }
}

const mapDispatchToProps = (dispatch) => ({
    addExpense: (expense) => dispatch(addExpense(expense))
});

export default connect(undefined, mapDispatchToProps)(AddExpensePage);