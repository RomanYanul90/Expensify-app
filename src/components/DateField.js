import React from 'react'

export const DateField = (props) => (
    <div>
        <h3>{props.createdAt}</h3>
    </div>

)
