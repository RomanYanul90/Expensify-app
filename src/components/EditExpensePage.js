import React from "react";
import {connect} from 'react-redux'
import Expense_form from "./Expense_form";
import {editExpense, startRemoveExpense} from "../actions/expenses";

export class EditExpensePage extends React.Component{
    onSubmit = (expense) => {
        this.props.editExpense(this.props.expense.id,expense)
        this.props.history.push('/')
    }
    onRemove = () => {
        this.props.startRemoveExpense({id:this.props.expense.id})
        this.props.history.push('/')
    }
    render(){
            return(
                <div>
                <Expense_form
                    expense = {this.props.expense}
                    onSubmit = {this.onSubmit}/>
            <button onClick = {this.onRemove}>Remove</button>
        </div>
            )
        }
}

const mapDispatchToProps =(dispatch)=>({
    editExpense: (id,expense) => dispatch(editExpense(id,expense)),
    startRemoveExpense:(data)=>dispatch(startRemoveExpense(data))
})

const mapStateToProps = (state,props)=>({expense: state.expenses.find((el)=>el.id===props.match.params.id)});

export default connect(mapStateToProps,mapDispatchToProps)(EditExpensePage)