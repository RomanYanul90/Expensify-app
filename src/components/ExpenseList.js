import React from 'react'
import {connect} from 'react-redux'
import ExpenseListItem from "./ExpenseListItem";
import selectExpenses from '../selectors/expenses'

export const ExpenseList=(props)=>(
<div>
    <h1>Expense List</h1>
    {
        props.expenses.length ===0 ? (
            <p>No expenses</p>
        ):(
            props.expenses.map((el)=>{
                return <ExpenseListItem key = {el.id} {...el}/>
            })
        )
    }

</div>
)

const mapStateToProps=(state)=>{
    return {
        expenses: selectExpenses(state.expenses,state.filters)
    }
}

const ConnectedExpenseList=connect(mapStateToProps)(ExpenseList)

export default ConnectedExpenseList