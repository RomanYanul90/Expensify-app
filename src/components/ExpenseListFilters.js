import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters';

export class ExpenseListFilters extends React.Component {
    onTexChange = (e) => {
        this.props.setTextFilter(e.target.value);
    }
    onSortChange = (e) => {
        if (e.target.value === 'date') {
            this.props.sortByDate();
        } else if (e.target.value === 'amount') {
            this.props.sortByAmount();
        }
    }
    render() {
        return (
            <div>
                <input
                    type = "text"
                    value = {this.props.filters.text}
                    onChange = {this.onTexChange}
                />
                <select
                    value={this.props.filters.sortBy}
                    onChange={this.onSortChange}
                >
                    <option value="date">Date</option>
                    <option value="amount">Amount</option>
                </select>
            </div>
        );
    }
}

const mapStateToProps = (state) => ({filters: state.filters});

const mapDispatchToProps = (dispatch) => ({
    setTextFilter:(text)=> dispatch(setTextFilter(text)),
    sortByAmount:()=>dispatch(sortByAmount()),
    sortByDate:()=>dispatch(sortByDate()),
    //setStartDate:(startDate)=>dispatch(setStartDate(startDate)),
    //setEndDate:(endDate)=>dispatch(setEndDate(endDate))
});

export default connect(mapStateToProps,mapDispatchToProps)(ExpenseListFilters);