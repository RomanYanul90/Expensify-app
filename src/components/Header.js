import {NavLink} from "react-router-dom";
import React from "react";

const Header=()=>(
    <header>
        <h1>Expensify</h1>
        <NavLink to='/' activeClassName='isActive' exact={true}>Dashboard</NavLink>
        <NavLink to='/create' activeClassName='isActive'>Create page</NavLink>
        <NavLink to='/help' activeClassName='isActive'>Help page</NavLink>
    </header>
);

export default Header