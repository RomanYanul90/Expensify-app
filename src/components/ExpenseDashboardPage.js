import React from "react";
import ExpenseList from "./ExpenseList";
import ExpenseListFilters from "./ExpenseListFilters";
import ExpensesSummary from "./ExpensesSummary";

const ExpenseDashboardPage=()=>(
    <h3>
        <ExpensesSummary/>
        <ExpenseList/>
        <ExpenseListFilters/>
    </h3>
);

export default ExpenseDashboardPage


