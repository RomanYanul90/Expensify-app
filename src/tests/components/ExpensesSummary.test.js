import React from 'react'
import {ExpensesSummary} from "../../components/ExpensesSummary";
import { shallow } from 'enzyme'

test('should render ExpenseSummary with 1 expense correctly',()=>{
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={1123}/>)
    expect(wrapper).toMatchSnapshot()
})

test('should render ExpenseSummary with more than one expenses correctly',()=>{
    const wrapper = shallow(<ExpensesSummary expenseCount={3} expensesTotal={1123423}/>)
    expect(wrapper).toMatchSnapshot()
})