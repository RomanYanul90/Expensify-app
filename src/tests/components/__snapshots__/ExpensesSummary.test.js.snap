// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary with 1 expense correctly 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling: 
    $11.23
  </h1>
</div>
`;

exports[`should render ExpenseSummary with more than one expenses correctly 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totalling: 
    $11,234.23
  </h1>
</div>
`;
